File: go/problem27.go
package main

import (
	"fmt"
	"math"
)

func main() {
	fmt.Println(QuadraticPrime(100))
}

func QuadraticPrime(limit int) int {

	maxA, maxB, maxN := 0, 0, 0

	for a := -limit; a <= limit; a++ {

		for b := -limit; b <= limit; b++ {

			count := 0
			for count < b {
				if isPrime(quadraticVal(a, b, count)) {
					count++
				} else {
					break
				}
			}

			if count > maxN {
				maxN = count
				maxA = a
				maxB = b
			}
		}
	}

	return maxA * maxB

}

func quadraticVal(a, b, n int) int {
	return int(math.Abs(float64(n*n + a*n + b)))
}

func isPrime(number int) bool {

	if number <= 1 {
		return false
	}

	if number%2 == 0 {
		return false
	}

	for i := 3; i <= int(math.Sqrt(float64(number))); i += 2 {
		if number%i == 0 {
			return false
		}
	}

	return true
}

File: python/problem27.py
import math

def quadraticPrime(limit):
    maxA, maxB, maxN = 0, 0, 0

    for a in range(-limit, limit):

        for b in range(-limit, limit):

            count = 0
            while count < b:
                if isPrime(quadraticVal(a, b, count)):
                    count += 1
                else:
                    break

            if count > maxN:
                maxN = count
                maxA = a
                maxB = b

    return maxA * maxB


def quadraticVal(a, b, n):
    return int(abs(n * n + a * n + b))


def isPrime(number):
    if number <= 1:
        return False

    if number % 2 == 0:
        return False

    for i in range(3, int(math.sqrt(number)), 2):
        if number % i == 0:
            return False

    return True


if __name__ == "__main__":
    print(quadraticPrime(100))
